components/Buttons/src/MDCButton.m

```diff
@@ -305,7 +310,26 @@ - (void)layoutSubviews {
 }
 
 - (BOOL)pointInside:(CGPoint)point withEvent:(UIEvent *)event {
-  return CGRectContainsPoint(UIEdgeInsetsInsetRect(self.bounds, _hitAreaInsets), point);
+  // If there are custom hitAreaInsets, use those
+  if (!UIEdgeInsetsEqualToEdgeInsets(self.hitAreaInsets, UIEdgeInsetsZero)) {
+    return CGRectContainsPoint(
+        UIEdgeInsetsInsetRect(CGRectStandardize(self.bounds), self.hitAreaInsets),
+        point);
+  }
+
+  // If the bounds are smaller than the minimum touch target, produce a warning once
+  CGFloat width = CGRectGetWidth(self.bounds);
+  CGFloat height = CGRectGetHeight(self.bounds);
+  if (width < MDCButtonMinimumTouchTargetWidth || height < MDCButtonMinimumTouchTargetHeight) {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+      NSLog(@"Button touch target does not meet minimum size guidlines of (48, 48). Button: %@, "
+            @"Touch Target: %@",
+            [self description],
+            NSStringFromCGSize(CGSizeMake(width, height)));
+    });
+  }
+  return [super pointInside:point withEvent:event];
```

Comment:
If we are switching to [super pointInside:withEvent:] here, do we also want to in our implementation if we have custom hitAreaInsets above?

Alternatively, should this return CGRectContainsPoint()?

If there is cause for these two cases to be different, we should document why.